tests/rbd_mirror/test_rbd_snapshot_mirroring_with_upgrade.py

```diff
@@ -0,0 +1,245 @@
+    try:
+        ceph_cluster = kw.get("ceph_cluster")
+        mon_node = ceph_cluster.get_nodes("installer")[0]
+        rc = ceph_cluster.check_health(
+            "6.1",
+            client=mon_node,
+            timeout=300,
+        )
+
+        if rc != 0:
+            log.error("Ceph health not OK after adopting cluster to use cephadm")
+            return 1
+
+        client = kw.get("client")
+        client.exec_command(cmd="ceph osd set noout", sudo=True)
+        client.exec_command(cmd="ceph osd set noscrub", sudo=True)
+        client.exec_command(cmd="ceph osd set nodeep-scrub", sudo=True)
+
+        with parallel() as p:
+            p.spawn(
+                execute,
+                mod_file_name="tests.cephadm.test_cephadm_upgrade",
+                args=kw,
+                test_name="Upgrade cluster",
+                raise_exception=True,
+            )
+            for pool_type in kw.get("pool_types"):
+                p.spawn(
+                    run_io_verify_snap_schedule,
+                    pool_type=pool_type,
+                    raise_exception=True,
+                    **kw,
+                )
+
+        client.exec_command(cmd="ceph osd unset noout", sudo=True)
+        client.exec_command(cmd="ceph osd unset noscrub", sudo=True)
+        client.exec_command(cmd="ceph osd unset nodeep-scrub", sudo=True)
+    except Exception as e:
+        log.error(f"Cluster upgrade failed with error {e}")
+        return 1
+    return 0
+
+
+def run(**kw):
+    """CEPH-83574895 - Configure two-way rbd-mirror on Stand alone
+    CEPH cluster on image (with replicated and ec pools)with snapshot
+    based mirroring and perform upgrade
+    Pre-requisites :
+    We need atleast one client node with ceph-common, fio and rbd-nbd packages,
+    conf and keyring files in both clusters with snapshot based RBD mirroring
+    enabled between the clusters.
+    kw:
+        clusters:
+            ceph-rbd1:
+            config:
+                rep_pool_config:
+                num_pools: 1
+                num_images: 5
+                size: 10G
+                mode: image # compulsory argument if mirroring needs to be setup
+                mirrormode: snapshot
+                snap_schedule_levels:
+                    - image
+                snap_schedule_intervals: #one value for each level specified above
+                    - 5m
+                io_percentage: 30 #percentage of space in each image to be filled
+                ec_pool_config:
+                num_pools: 1
+                num_images: 5
+                mode: image # compulsory argument if mirroring needs to be setup
+                mirrormode: snapshot
+                snap_schedule_levels:
+                    - image
+                snap_schedule_intervals:
+                    - 5m
+                io_percentage: 30
+                command: start
+                service: upgrade
+                verify_cluster_health: true
+            ceph-rbd1:
```

Comment:
please remove the redundant config example one config example is enough.

Reply:
its one config for each cluster, i have changed it correctly now.